From 69468242f04eb2bc4044abe9d48a90516f1ef387 Mon Sep 17 00:00:00 2001
From: Wayne Zou <b36644@freescale.com>
Date: Wed, 2 May 2012 14:59:01 +0800
Subject: ENGR00178630-2 MX6DL SabreSD: Add enable_lcd_ldb cmdline option for
 LVDS+WVGA

Add enable_lcd_ldb command line option for dual display
with LVDS and WVGA LCD panel together on MX6DL.

Signed-off-by: Wayne Zou <b36644@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabresd.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabresd.c b/arch/arm/mach-mx6/board-mx6q_sabresd.c
index d1ac2f4..83746e9 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabresd.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabresd.c
@@ -201,6 +201,7 @@ static struct clk *sata_clk;
 static int mma8451_position = 3;
 static int mag3110_position;
 static int max11801_mode = 1;
+static int enable_lcd_ldb;
 
 
 extern char *gp_reg_id;
@@ -1523,6 +1524,13 @@ static const struct imx_pcie_platform_data mx6_sabresd_pcie_data __initconst = {
 	.pcie_dis	= SABRESD_PCIE_DIS_B,
 };
 
+static int __init early_enable_lcd_ldb(char *p)
+{
+	enable_lcd_ldb = 1;
+	return 0;
+}
+early_param("enable_lcd_ldb", early_enable_lcd_ldb);
+
 /*!
  * Board specific initialization.
  */
@@ -1570,6 +1578,10 @@ static void __init mx6_sabresd_board_init(void)
 		hdmi_core_data.disp_id = 1;
 		mipi_dsi_pdata.ipu_id = 0;
 		mipi_dsi_pdata.disp_id = 1;
+		if (enable_lcd_ldb) {
+			ldb_data.disp_id = 1;
+			ldb_data.mode = LDB_SIN1;
+		}
 	}
 	imx6q_add_mxc_hdmi_core(&hdmi_core_data);
 
-- 
1.7.9.5

